src/Components/OATHeader/internal/FormOpen.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from 'react';
+import {
+    Text,
+    ActionButton,
+    FontIcon,
+    PrimaryButton,
+    Stack,
+    Dropdown
+} from '@fluentui/react';
+import { useTranslation } from 'react-i18next';
+import { IAction } from '../../../Models/Constants/Interfaces';
+import {
+    OATDataStorageKey,
+    OATFilesStorageKey
+} from '../../../Models/Constants';
+import {
+    SET_OAT_PROJECT_NAME,
+    SET_OAT_PROPERTY_EDITOR_MODEL,
+    SET_OAT_RELOAD_PROJECT
+} from '../../../Models/Constants/ActionTypes';
+import { getHeaderStyles } from '../OATHeader.styles';
+
+interface IModal {
+    dispatch?: React.Dispatch<React.SetStateAction<IAction>>;
+    setModalBody?: React.Dispatch<React.SetStateAction<string>>;
+    setModalOpen?: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const FormOpen = ({ dispatch, setModalOpen, setModalBody }: IModal) => {
+    const { t } = useTranslation();
+    const headerStyles = getHeaderStyles();
+    const [files, setFiles] = useState([]);
+    const [selectedFile, setSelectedFile] = useState(null);
+
+    const handleOpen = () => {
+        dispatch({
+            type: SET_OAT_PROPERTY_EDITOR_MODEL,
+            payload: null
+        });
+
+        dispatch({
+            type: SET_OAT_PROJECT_NAME,
+            payload: selectedFile.text
+        });
+
+        dispatch({
+            type: SET_OAT_RELOAD_PROJECT,
+            payload: true
+        });
```

Comment:
I wonder if we want to create a new operation to dispatch that sets all three of these?